tests/AutomationTests/Services/YamlFileServiceTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Oryx.Automation.Models;
+using Microsoft.Oryx.Automation.Services;
+
+namespace Microsoft.Oryx.Automation.Tests.Services
+{
+    public class YamlFileServiceTests
+    {
+        private readonly YamlFileService yamlFileService;
+        private readonly string oryxRootPath;
+        private readonly string testConstantsYamlFilePath;
+
+        public YamlFileServiceTests()
+        {
+            this.oryxRootPath = Directory.GetCurrentDirectory();
+            this.testConstantsYamlFilePath = Path.Combine(this.oryxRootPath, "test-constants.yaml");
+            this.yamlFileService = new YamlFileService(this.oryxRootPath);
+        }
+
+        [Fact]
+        public async Task ReadConstantsYamlFileAsync_ReturnsExpectedYamlContents()
+        {
+            // Arrange
+            string yamlContents = "- name: test\r\n  constants:\r\n    key1: value1\r\n    key2: value2";
+            File.WriteAllText(this.testConstantsYamlFilePath, yamlContents);
```

Comment:
For the tests in this file where we're writing the contents to a new file, we should handle the case where an unexpected error is thrown but we still delete the file we generated. In this case, if the `ReadConstantsYamlFileAsync` method below fails for an unexpected error, we never hit the `File.Delete` call at the bottom and the file we wrote to lingers around (which isn't a _huge_ deal since these run on separate container agents, but locally it may linger).

I suggest wrapping the test with a `try-finally`:

```cs
try
{
    // Arrange
   ...

    // Act
    ...

    // Assert
    ...
}
finally
{
    // Clean up
    File.Delete(this.testConstantsYamlFilePath);
}
```

Reply:
Great idea! Especially since we only have 10GB disk space in our agents.